### ISSUE TITLE
AttributeError: module 'locale' has no attribute 'LC_MESSAGES'

### ISSUE THREAD

**Issue by bobhy** [xonsh/xonsh#3666]
Seems like very recent checkins to `xonsh/environ.py` may have destabilized xonsh on Windows (where `locale` package has no `LC_MESSAGES`).

## xonfig

<details>
(pip list, can't run `xonfig`)

```
(.venv) PS C:\Users\bobhy\src\xonsh> python
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()

(.venv) PS C:\Users\bobhy\src\xonsh> pip list
Package               Version
--------------------- ---------
appdirs               1.4.4
atomicwrites          1.4.0
attrs                 19.3.0
black                 19.10b0
certifi               2020.6.20
cfgv                  3.2.0
chardet               3.0.4
click                 7.1.2
codecov               2.1.8
colorama              0.4.3
coverage              5.2.1
distlib               0.3.1
filelock              3.0.12
flake8                3.8.3
identify              1.4.25
idna                  2.10
iniconfig             1.0.1
mccabe                0.6.1
more-itertools        8.4.0
nodeenv               1.4.0
packaging             20.4
pathspec              0.8.0
pip                   20.2.1
pluggy                0.13.1
pre-commit            2.6.0
prompt-toolkit        3.0.5
py                    1.9.0
pycodestyle           2.6.0
pyflakes              2.2.0
Pygments              2.6.1
pyparsing             2.4.7
pytest                6.0.1
pytest-cov            2.10.0
pytest-flake8         1.0.6
pytest-timeout        1.4.2
PyYAML                5.3.1
regex                 2020.7.14
requests              2.24.0
setuptools            47.1.0
six                   1.15.0
toml                  0.10.1
typed-ast             1.4.1
urllib3               1.25.10
virtualenv            20.0.30
wcwidth               0.2.5
xonsh                 0.9.18
xonsh-apt-tabcomplete 0.1.6
```

</details>

## Expected Behavior
Master branch should at least initialize on all supported platforms.  (Though I hesitate to cast the first stone...)
## Current Behavior
Xonsh fails on startup on Windows 10.

### Traceback (if applicable)

<details>

```
(.venv) PS C:\Users\bobhy\src\xonsh> xonsh.bat
Traceback (most recent call last):
  File "C:\Program Files\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\bobhy\src\xonsh\xonsh\__main__.py", line 3, in <module>
    main()
  File "C:\Users\bobhy\src\xonsh\xonsh\main.py", line 426, in main
    _failback_to_other_shells(args, err)
  File "C:\Users\bobhy\src\xonsh\xonsh\main.py", line 395, in _failback_to_other_shells
    raise err
  File "C:\Users\bobhy\src\xonsh\xonsh\main.py", line 423, in main
    args = premain(argv)
  File "C:\Users\bobhy\src\xonsh\xonsh\main.py", line 359, in premain
    env = start_services(shell_kwargs, args)
  File "C:\Users\bobhy\src\xonsh\xonsh\main.py", line 285, in start_services
    execer = Execer(
  File "C:\Users\bobhy\src\xonsh\xonsh\execer.py", line 63, in __init__
    load_builtins(execer=self, ctx=xonsh_ctx)
  File "C:\Users\bobhy\src\xonsh\xonsh\built_ins.py", line 1304, in load_builtins
    builtins.__xonsh__.load(execer=execer, ctx=ctx)
  File "C:\Users\bobhy\src\xonsh\xonsh\built_ins.py", line 1376, in load
    self.env = Env(default_env())
  File "C:\Users\bobhy\src\xonsh\xonsh\environ.py", line 2168, in default_env
    ctx = dict(BASE_ENV)
  File "C:\Users\bobhy\src\xonsh\xonsh\lazyasd.py", line 62, in __iter__
    obj = self._lazy_obj()
  File "C:\Users\bobhy\src\xonsh\xonsh\lazyasd.py", line 46, in _lazy_obj
    obj = d["load"]()
  File "C:\Users\bobhy\src\xonsh\xonsh\environ.py", line 2096, in <lambda>
    "BASH_COMPLETIONS": list(DEFAULT_VARS["BASH_COMPLETIONS"].default),
  File "C:\Users\bobhy\src\xonsh\xonsh\lazyasd.py", line 66, in __getitem__
    obj = self._lazy_obj()
  File "C:\Users\bobhy\src\xonsh\xonsh\lazyasd.py", line 46, in _lazy_obj
    obj = d["load"]()
  File "C:\Users\bobhy\src\xonsh\xonsh\environ.py", line 993, in DEFAULT_VARS
    locale.setlocale(locale.LC_MESSAGES),
AttributeError: module 'locale' has no attribute 'LC_MESSAGES'

```

</details>

## Steps to Reproduce
(On Windows 10)
```
$ git pull master
$ python setup.py clean install
$ xonsh.bat
```

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by melund** [comment]
There is already a PR to fix this https://github.com/xonsh/xonsh/pull/3658

**Comment by laloch** [comment]
Closed in #3658.
